src/app.md

```diff
@@ -271,6 +272,54 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-file-skip-dot] #ex-file-skip-dot
+<a name="ex-file-skip-dot"></a>
+##  Traverse directories while skipping dotfiles
+
+Iterate over the entries that are not hidden files or hidden directories. 
+
+[`WalkDir::filter_entry`] descend recursively entries passing the predicate whereas [`std::iter::Iterator::filter`] would by applied to each `DirEntry` even if the parent is a hidden directory. [`WalkDir::depth`] is used to skip the level 0 which is the current directory `"."`.
+
+[![walkdir-badge]][walkdir] [![cat-filesystem-badge]][cat-filesystem]
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate walkdir;
+
+use walkdir::{DirEntry, WalkDir, WalkDirIterator};
+#
+# error_chain! {
+#     foreign_links {
+#         WalkDir(walkdir::Error);
+#         Io(std::io::Error);
+#     }
+# }
+#
+
+fn is_not_hidded(entry: &DirEntry) -> bool {
+    !entry
+         .file_name()
+         .to_str()
+         .map(|s| entry.depth() > 0 && s.starts_with("."))
+         .unwrap_or(false)
+}
+
+fn run() -> Result<()> {
```

Comment:
Well in rust it's a little more mundane and yet a little more magical at the same time. In fact you cannot return more than one error type instead your error type has to implement `from` for each error variant (actually there is a little more painful [boilerplate](https://rustbyexample.com/error/define_error_type.html) there that error-chain solves with macros). This is resolved at compiletime instead of runtime like in java. 

In case you are missing a variant you will get error like `from not implemented for <missing_error_type>`.
I hope this helps a little